report_xlsx/controllers/main.py

```diff
@@ -3,39 +3,45 @@
 import time
+import werkzeug
 
 
 class ReportController(report.ReportController):
     @route()
     def report_routes(self, reportname, docids=None, converter=None, **data):
         if converter == 'xlsx':
-            report = request.env['ir.actions.report']._get_report_from_name(
-                reportname)
-            context = dict(request.env.context)
-            if docids:
-                docids = [int(i) for i in docids.split(',')]
-            if data.get('options'):
-                data.update(json.loads(data.pop('options')))
-            if data.get('context'):
-                # Ignore 'lang' here, because the context in data is the one
-                # from the webclient *but* if the user explicitely wants to
-                # change the lang, this mechanism overwrites it.
-                data['context'] = json.loads(data['context'])
-                if data['context'].get('lang'):
-                    del data['context']['lang']
-                context.update(data['context'])
-            xlsx = report.with_context(context).render_xlsx(
-                docids, data=data
-            )[0]
-            report_name = report.report_file
-            if report.print_report_name and not len(docids) > 1:
-                obj = request.env[report.model].browse(docids[0])
-                report_name = safe_eval(report.print_report_name,
-                                        {'object': obj, 'time': time})
+            try:
+                report = request.env['ir.actions.report']._get_report_from_name(
+                    reportname)
+                context = dict(request.env.context)
+                if docids:
+                    docids = [int(i) for i in docids.split(',')]
+                if data.get('options'):
+                    data.update(json.loads(data.pop('options')))
+                if data.get('context'):
+                    # Ignore 'lang' here, because the context in data is the one
+                    # from the webclient *but* if the user explicitely wants to
+                    # change the lang, this mechanism overwrites it.
+                    data['context'] = json.loads(data['context'])
+                    if data['context'].get('lang'):
+                        del data['context']['lang']
+                    context.update(data['context'])
+                xlsx = report.with_context(context).render_xlsx(
+                    docids, data=data
+                )[0]
+                report_name = report.report_file
+                if report.print_report_name and not len(docids) > 1:
+                    obj = request.env[report.model].browse(docids[0])
+                    report_name = safe_eval(report.print_report_name,
+                                            {'object': obj, 'time': time})
+            except (UserError, ValidationError) as odoo_error:
+                raise werkzeug.exceptions.HTTPException(
+                    description=f'{odoo_error.name}. {odoo_error.value}')
```

Comment:
Functional Test.
Case `Exception` is not showed error.
you should coding `except Exception as e :` it's cover all case.

Reply:
```except Exception as e``` is too generic and a bad pratic. I need to intercept only user error so if we use UserError or ValidationError in our code, the messages can be showed. All the others errors (implementation errors) can be found in odoo server log.

Reply:
okay, I see.